File: src/main/java/org/example/expert/domain/todo/dto/response/TodoListRespDto.java
package org.example.expert.domain.todo.dto.response;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.example.expert.domain.todo.entity.Todo;
import org.example.expert.domain.user.dto.response.UserResponse;
import org.springframework.data.domain.Page;

import java.util.List;

@NoArgsConstructor
@Getter
public class TodoListRespDto {

    private List<TodoResponse> todoList;

    private long totalElements;
    private int totalPages;
    private int pageNumber;
    private boolean hasNext;
    private boolean hasPrevious;

    public TodoListRespDto(Page<Todo> todoPageList) {
        this.todoList = todoPageList.getContent()
                .stream()
                .map(TodoResponse::new)
                .toList();
        this.totalElements = todoPageList.getTotalElements();
        this.totalPages = todoPageList.getTotalPages();
        this.pageNumber = todoPageList.getNumber();
        this.hasNext = todoPageList.hasNext();
        this.hasPrevious = todoPageList.hasPrevious();
    }
}

File: src/main/java/org/example/expert/controller/todo/dto/response/TodoListRespDto.kt
package org.example.expert.controller.todo.dto.response

import org.example.expert.domain.todo.Todo
import org.springframework.data.domain.Page

data class TodoListRespDto(
    val todoList: List<TodoInfoRespDto> = listOf(),
    val totalElements: Long,
    val pageNumber: Int,
    val totalPages: Int,
    val hasNext: Boolean,
    val hasPrevious: Boolean
) {
    constructor(todoPage: Page<Todo>): this(
        todoList = todoPage.content.map { TodoInfoRespDto(it) },
        totalElements = todoPage.totalElements,
        totalPages = todoPage.totalPages,
        pageNumber = todoPage.number,
        hasNext = todoPage.hasNext(),
        hasPrevious = todoPage.hasPrevious()
    )
}